File: python/1.py
import unittest
from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapNumToPos: Dict[int, int] = {}
        for idx in range(len(nums)):
            if target - nums[idx] in mapNumToPos:
                return [mapNumToPos[target - nums[idx]], idx]
            mapNumToPos[nums[idx]] = idx
        return []


class TestSolution(unittest.TestCase):
    def test(self):
        solution = Solution()
        self.assertListEqual(solution.twoSum([2, 7, 11, 15], 9), [0, 1])
        self.assertListEqual(solution.twoSum([3, 2, 4], 6), [1, 2])
        self.assertListEqual(solution.twoSum([3, 3], 6), [0, 1])

File: typescript/1.ts
function twoSum(nums: number[], target: number): number[] {
    const mapNumToPos: Map<number, number> = new Map();
    for (let i = 0; i < nums.length; i++) {
        if (mapNumToPos.has(target - nums[i])) {
            return [mapNumToPos.get(target - nums[i])!, i];
        }
        mapNumToPos.set(nums[i], i);
    }
    return [];
};

test('1', () => {
    expect(twoSum([2, 7, 11, 15], 9)).toStrictEqual([0, 1]);
});